src/app/firebase/services/firebase-player.service.spec.ts

```diff
@@ -1,21 +1,33 @@
 import { FirebasePlayerService } from './firebase-player.service';
-import { AngularFirestore } from '@angular/fire/compat/firestore';
+import { AngularFirestore, AngularFirestoreCollection, DocumentReference } from '@angular/fire/compat/firestore';
 import { Store } from '@ngrx/store';
+import { Player } from '../../player-creation/models/player';
 import { of } from 'rxjs';
 
 describe('FirebasePlayerService', () => {
   let service: FirebasePlayerService;
-  let fireStore: AngularFirestore;
+  let firestore: AngularFirestore;
   let store: Store;
 
   beforeEach(() => {
-    const document = { valueChanges: () => of({}) };
-    fireStore = { collection: _ => document } as AngularFirestore;
+    firestore = { collection: _ => {} } as AngularFirestore;
     store = { select: (_: any) => {} } as Store;
-    service = new FirebasePlayerService(fireStore, store);
   });
 
-  it('should be created', () => {
-    expect(service).toBeTruthy();
+  it('adds player to firebase', () => {
+    const collection = { add: _ => Promise.resolve({ id: 'playerId' } as DocumentReference<Player>) } as AngularFirestoreCollection<Player>;
+    spyOn(firestore, 'collection').and.returnValue(collection);
+    service = new FirebasePlayerService(firestore, store);
+
+    service.add({} as Player).subscribe(id => expect(id).toEqual('playerId'));
+  });
+
+  it('returns valueChanges of collection', () => {
```

Comment:
I don't like valueChanges :(

Reply:
I called it that because firebase calls it that too. I am open for suggestions as always (but not in this PR HAHAHAHA)